langchain/src/agents/tools/base.ts

```diff
@@ -1,5 +1,41 @@
+import { CallbackManager, getCallbackManager } from "../../callbacks/index.js";
+
+const getVerbosity = () => false;
+
+export interface ToolParams {
+  verbose?: boolean;
+  callbackManager?: CallbackManager;
+}
+
 export abstract class Tool {
-  abstract call(arg: string): Promise<string>;
+  verbose: boolean;
+
+  callbackManager: CallbackManager;
+
+  constructor(verbose?: boolean, callbackManager?: CallbackManager) {
+    this.verbose = verbose ?? getVerbosity();
+    this.callbackManager = callbackManager ?? getCallbackManager();
+  }
+
+  protected abstract _call(arg: string): Promise<string>;
```

Comment:
All tools now implement abstract `_call` instead of `call` s.t. we are able to plug in callbacks